e2e_tests/tests/cluster/test_users.py

```diff
@@ -203,346 +95,199 @@ def test_post_user_api(clean_auth: None, login_admin: None) -> None:
-    expected = "Unauthenticated"
-    assert expected in str(child.read())
-    child.wait()
-    child.close()
-    assert child.status != 0
-
-    # Log in as determined.
-    api_utils.configure_token_store(
-        authentication.Credentials(constants.DEFAULT_DETERMINED_USER, password)
-    )
-
-    # Log back in as new user.
-    api_utils.configure_token_store(creds)
+def test_logout() -> None:
+    # Make sure that a logged out session cannot be reused.
+    sess = api_utils.make_session("determined", "")
 
-    # Now log out as determined.
-    log_out_user(constants.DEFAULT_DETERMINED_USER)
-
-    # Should still be able to list experiments because new user is logged in.
-    child = det_spawn(["e", "list"])
-    child.read()
-    child.wait()
-    child.close()
-    assert child.status == 0
-
-    # Change Determined password back to "".
-    change_user_password(constants.DEFAULT_DETERMINED_USER, "")
-    # Clean up.
+    bindings.post_Logout(sess)
+    with pytest.raises(errors.UnauthenticatedException):
+        bindings.get_GetMe(sess)
 
 
 @pytest.mark.e2e_cpu
 @pytest.mark.e2e_cpu_postgres
-def test_activate_deactivate(clean_auth: None, login_admin: None) -> None:
-    creds = api_utils.create_test_user(True)
-
-    # Make sure we can log in as the user.
-    api_utils.configure_token_store(creds)
-
-    # Log out.
-    log_out_user()
-
-    # login admin again.
-    api_utils.configure_token_store(conf.ADMIN_CREDENTIALS)
+def test_activate_deactivate() -> None:
+    sess, password = api_utils.create_test_user(True)
 
     # Deactivate user.
-    activate_deactivate_user(False, creds.username)
+    admin = api_utils.admin_session()
+    activate_deactivate_user(admin, False, sess.username)
 
-    # Attempt to log in again. It should have a non-zero exit status.
-    log_in_user_cli(creds, 1)
+    # Attempt to log in again.
+    with pytest.raises(errors.ForbiddenException):
+        api_utils.make_session(sess.username, password)
 
     # Activate user.
-    activate_deactivate_user(True, creds.username)
+    activate_deactivate_user(admin, True, sess.username)
 
-    # Now log in again. It should have a non-zero exit status.
-    api_utils.configure_token_store(creds)
+    # Now log in again.
+    api_utils.make_session(sess.username, password)
 
     # SDK testing for activating and deactivating.
-    api_utils.configure_token_store(conf.ADMIN_CREDENTIALS)
-    det_obj = Determined(master=conf.make_master_url())
-    user = det_obj.get_user_by_name(user_name=creds.username)
+    det_obj = client.Determined._from_session(admin)
+    user = det_obj.get_user_by_name(user_name=sess.username)
     user.deactivate()
     assert user.active is not True
+    with pytest.raises(errors.ForbiddenException):
+        api_utils.make_session(sess.username, password)
+
     user.activate()
     assert user.active is True
-
-    # Now log in again.
-    api_utils.configure_token_store(creds)
+    api_utils.make_session(sess.username, password)
```

Comment:
yes, that there isn't an exception.  I would argue that running it _is_ explicitly validating that it doesn't raise an exception.

What did you have in mind here?

Reply:
I was mostly imagining some kind of simple assert on the session. It's mostly a nit and me thinking out loud while reading the PR. It would probably be clear enough what the issue was if this test were to fail.